cordova-lib/src/cordova/plugin.js

```diff
@@ -154,6 +135,38 @@ module.exports = function plugin(command, targets, opts) {
                             is_top_level: true
                         };
 
+                        // If no version is specified, retrieve the version (or source) from config.xml
+                        if (!version && !cordova_util.isUrl(id) && !cordova_util.isDirectory(id)) {
+                            events.emit('verbose', 'no version specified, retrieving version from config.xml');
+                            var ver = getVersionFromConfigFile(id, cfg);
+
+                            if (cordova_util.isUrl(ver) || cordova_util.isDirectory(ver)) {
+                                target = ver;
+                            } else if (ver) {
+                                // If version exists in config.xml, use that
+                                target = ver ? (id + '@' + ver) : target;
```

Comment:
I know this isn't your code - but would be good to clean this up while we're here... We already know `ver` has a value (since that is the condition of the `if` statement), so this should just be `target = id + '@' + ver`.